scheduler/src/cook/mesos.clj

```diff
@@ -194,43 +157,39 @@
                               ;; TODO: get the framework ID and try to reregister
                               (let [normal-exit (atom true)]
                                 (try
-                                  (let [{:keys [scheduler view-incubating-offers]}
+                                  (let [{:keys [pool-name->fenzo pool->offers-chan view-incubating-offers]}
                                         (sched/create-datomic-scheduler
                                          {:conn mesos-datomic-conn
                                           :compute-cluster compute-cluster
-                                          :exit-code-syncer-state exit-code-syncer-state
                                           :fenzo-fitness-calculator fenzo-fitness-calculator
                                           :fenzo-floor-iterations-before-reset fenzo-floor-iterations-before-reset
                                           :fenzo-floor-iterations-before-warn fenzo-floor-iterations-before-warn
                                           :fenzo-max-jobs-considered fenzo-max-jobs-considered
                                           :fenzo-scaleback fenzo-scaleback
                                           :good-enough-fitness good-enough-fitness
-                                          :gpu-enabled? gpu-enabled?
-                                          :heartbeat-ch mesos-heartbeat-chan
                                           :mea-culpa-failure-limit mea-culpa-failure-limit
                                           :mesos-run-as-user mesos-run-as-user
                                           :agent-attributes-cache agent-attributes-cache
                                           :offer-incubate-time-ms offer-incubate-time-ms
                                           :pool-name->pending-jobs-atom pool-name->pending-jobs-atom
-                                          :progress-config progress-config
                                           :rebalancer-reservation-atom rebalancer-reservation-atom
-                                          :sandbox-syncer-state sandbox-syncer-state
                                           :task-constraints task-constraints
                                           :trigger-chans trigger-chans})
-                                        driver (make-mesos-driver-fn scheduler framework-id)]
-                                    (mesomatic.scheduler/start! driver)
-                                    (reset! current-driver driver)
-                                    (cc/set-mesos-driver-atom-hack! compute-cluster driver)
-
+                                        cluster-leadership-promise (cc/initialize-cluster compute-cluster
```

Comment:
Something like 'We leave a thread blocked .takeLeadership. When that thread is unblocked by curator, it will subsequently send a message on this channel, unblocking us here."

Reply:
Did you see the comment below?